metrics/metrics/src/main/java/io/helidon/metrics/HelidonFunctionalCounter.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.metrics;
+
+import java.util.function.ToDoubleFunction;
+
+import io.micrometer.core.instrument.FunctionCounter;
+import io.micrometer.core.instrument.MeterRegistry;
+import org.eclipse.microprofile.metrics.Counter;
+import org.eclipse.microprofile.metrics.Metadata;
+import org.eclipse.microprofile.metrics.Tag;
+
+/**
+ * Implementation of a counter wrapper around an object capable of providing a double value.
+ *
+ * @param <T> type of the object providing the value
+ */
+class HelidonFunctionalCounter<T> extends MetricImpl implements Counter {
```

Comment:
Do I understand it correctly that this is basically a Gauge? Why do we still have it? Or is it represented differently by the format?

Reply:
Basically, this allows a `Counter`-style "wrapper" around an externally-updated value instead of a `Gauge`-style wrapper.

Earlier releases of MP metrics allowed developers to create their own implementations of metrics and register them. Helidon used this to register custom counters, for example. 

MP Metrics 5.0 no longer permits this, but the ability to have a `Counter`-like wrapper is useful (and present in Micrometer and OTel metrics) so will likely be in the Helidon neutral API as well.